sky/cloud_stores.py

```diff
@@ -130,6 +130,11 @@ def is_directory(self, url: str) -> bool:
         out = p.stdout.decode().strip()
         # Edge Case: Gcloud command is run for first time #437
         out = out.split('\n')[-1]
+        if 'CommandException: One or more URLs matched no objects' in out or \
+             'bucket does not exist' in out:
+            raise sky_exceptions.StorageBucketGetError(
+                f'{url} does not exist in Google Cloud Storage. '
+                '(Typo, perhaps?)')
```

Comment:
This code will not be reached if the command fails, since an exception gets thrown at 125. We may want to catch the exception there and move this check earlier up.

Reply:
I am terribly sorry for the late response. I simply can't figure this out. Is this how I should do it?
~~~python
        try:
            p = subprocess.run(command,
                           stdout=subprocess.PIPE,
                           shell=True,
                           check=True,
                           executable='/bin/bash')
        except subprocess.CalledProcessError as e:
            if 'CommandException: One or more URLs matched no objects' in e or \
             'bucket does not exist' in e:
                raise sky_exceptions.StorageBucketGetError(
                    f'{url} does not exist in Google Cloud Storage. '
                    '(Typo, perhaps?)')
            raise e
~~~